src/OpenSage.Game/Logic/SelectionSystem.cs

```diff
@@ -23,20 +24,32 @@ public enum SelectionStatus
         // This should probably scale with resolution.
         private const int BoxSelectionMinimumSize = 30;
 
+        // TODO: consider allowing configuration for accessibility
+        // time after which releasing right click will not result in deselecting units, in ms
+        private const int DeselectMaxTimeMs = 250;
+
+        // TODO: scale with resolution, allow configuration for accessibility
+        // if the cursor moves further than this, don't deselect no matter how short right click was held for
+        private const int DeselectMaxDelta = 5;
+
         private SelectionGui SelectionGui => Game.Scene3D.SelectionGui;
 
-        private Point2D _startPoint;
-        private Point2D _endPoint;
+        private Point2D _selectionStartPoint;
+        private Point2D _selectionEndPoint;
+
+        private Point2D _panStartPoint;
+        private readonly Stopwatch _panStopwatch = new Stopwatch();
```

Comment:
Looking through more of the solution, I'm actually not convinced using a stopwatch is the best course of action here. Should I be using `TimeInterval` and `Game.MapTime` instead? That seems to be the established pattern elsewhere.

Reply:
Indeed stopwatch is not the preferred way to do it.

Reply:
Yes, we don't use StopWatches anywhere usually

Reply:
I'll get that changed, thanks